### PR TITLE
[versionlock] Add --raw option (RhBug:1645564)

### PR DESCRIPTION
**Pull request by m-blaha**
Versionlock add/exclude commands are now able to insert into
versionlock.list file patterns without converting it into NEVRAs. It is
usable for example for locking package to versions not available yet:

$ dnf version lock --raw add 'bash-5.*'
Adding versionlock on: bash-5.*

This command will keep the bash package on major version 5 and dnf will
still consume any future updates as far as they do not bump the major
version.

Without --raw switch used, the versionlock would take only currently
available packages into account:

$ dnf versionlock add 'bash-5.*'
Adding versionlock on: bash-0:5.0.7-1.fc30.*

https://bugzilla.redhat.com/show_bug.cgi?id=1645564

### CODE REVIEW

File: doc/versionlock.rst [hunk 1]
```diff
@@ -117,6 +117,13 @@ Options
 
 All general DNF options are accepted, see `Options` in :manpage:`dnf(8)` for details.
 
+``--raw``
+    Do not parse ``<package-name-spec>`` to NEVRAs to find specific version to lock to. Instead ``<package-name-spec>`` are used as they are. This enables locking to versions of package not yet available.
```
Comment (lukash) [inline doc/versionlock.rst]: This reads a bit better IMO:
> ... This enables locking to not yet available versions of the package.

File: doc/versionlock.rst [hunk 2]
```diff
@@ -117,6 +117,13 @@ Options
...
+    For example you want to keep the `bash` package on major version 5 and consume any future updates as far as they keep the major version::
```
Comment (lukash) [inline doc/versionlock.rst]: > For example you **may** want to ...

**Comment by m-blaha** [comment]
Thanks @lukash, fixed.

**Comment by lukash** [comment]
@m-blaha I think we could use tests in `ci-dnf-stack` for the `--raw` switch. `versionlock` tests are there already, so adding this should be easy.

**Comment by m-blaha** [comment]
Yes, I'll add the tests.

**Comment by lukash** [comment]
Waiting for tests...

**Comment by m-blaha** [comment]
Here is PR with the tests: https://github.com/rpm-software-management/ci-dnf-stack/pull/681

File: doc/versionlock.rst [hunk 3]
```diff
@@ -117,6 +117,13 @@ Options
 
 All general DNF options are accepted, see `Options` in :manpage:`dnf(8)` for details.
 
+``--raw``
+    Do not parse ``<package-name-spec>`` to NEVRAs to find specific version to lock to. Instead ``<package-name-spec>`` are used as they are. This enables locking to not yet available versions of the package.
```
Comment (lukash) [inline doc/versionlock.rst]: Okay, one more minor thing, can you please replace _"parse"_ with _"resolve"_ here?
> Do not ** resolve** ``<package-name-spec>`` to NEVRAs ...

I think it better describes what's going to happen.

**Pull request merged**